src/runtime/graph_executor/graph_executor.cc

```diff
@@ -139,6 +156,30 @@ void GraphExecutor::SetInputZeroCopy(int index, DLTensor* data_ref) {
     t->data = data_ref->data;
   }
 }
+/*!
+ * \brief set index-th output to the graph without copying the data.
+ * \param index The output index.
+ * \param data_ref The output data that is referred.
+ */
+void GraphExecutor::SetOutputZeroCopy(int index, DLTensor* data_ref) {
+  ICHECK_LT(static_cast<size_t>(index), outputs_.size());
+  uint32_t eid = this->entry_id(outputs_[index]);
+  const DLTensor* old_t = data_entry_[eid].operator->();
+
+  // check the consistency of output
+  ICHECK_EQ(data_alignment_[eid], details::GetDataAlignment(*data_ref));
+  ICHECK_EQ(reinterpret_cast<size_t>(data_ref->data) % kAllocAlignment, 0);
+  ICHECK_EQ(old_t->ndim, static_cast<size_t>(data_ref->ndim));
+  ICHECK_EQ(old_t->device.device_type, data_ref->device.device_type);
+  ICHECK_EQ(old_t->device.device_id, data_ref->device.device_id);
+  for (auto i = 0; i < data_ref->ndim; ++i) {
+    ICHECK_EQ(old_t->shape[i], data_ref->shape[i]);
+  }
```

Comment:
SetInputZeroCopy use exactly same code, a common function  should can help to avoid  duplicate.

Reply:
I will modify